Replace RanChangeSetSerializer with wrapper class for serde compatibility

yawkat opened the pull request.

graemerocher commented on liquibase/src/main/java/io/micronaut/liquibase/endpoint/RanChangeSetWrapper.java:
```diff
@@ -0,0 +1,93 @@
+import java.util.Set;
+
+@Introspected
+final class RanChangeSetWrapper {
```
Add `@Internal` and `@since`

ilopmar commented on liquibase/src/main/java/io/micronaut/liquibase/endpoint/RanChangeSetSerializer.java:
```diff
@@ -1,64 +0,0 @@
- * @author Iván López
- * @since 1.0.0
- */
-@Singleton
-@Type(RanChangeSet.class)
-public class RanChangeSetSerializer extends JsonSerializer<RanChangeSet> {
```
Removing a public class is a breaking change. I doubt that someone is using it... but :shrug:

ilopmar reviewed (changes requested):
The output is not the same as before (image on the right). You can see that there are a few missing fields and the name of `checksum` is now `lastChecksum`:

![image](https://user-images.githubusercontent.com/559192/143887153-e5dae7e3-30c1-4bdd-9359-bccde1aebe53.png)

yawkat commented:
Thanks, i've added a test to verify that all properties are present as they should be, and fixed it.

Also readded the serializer class, it doesn't hurt serde as long as nobody uses it.

ilopmar reviewed (approved):
Looks good. The only thing I'm not really sure about if these changes need to go in `5.0.1` or in `5.1.0`.

@jameskleeh thoguths?

jameskleeh commented:
Deprecations can only be done in minor releases

ilopmar commented:
@yawkat please change version to `5.1.0` and target micronaut core `3.3.x`

yawkat commented:
done

ilopmar commented on liquibase/src/main/java/io/micronaut/liquibase/endpoint/RanChangeSetWrapper.java:
```diff
@@ -0,0 +1,103 @@
+/**
+ * Serialization wrapper around {@link RanChangeSet}.
+ *
+ * @author Jonas Konrad
+ * @since 5.0.1
```
@yawkat you missed this `@since` :pray:

yawkat commented:
nice catch :)
